src/tools/miri/tests/pass-dep/shims/fs.rs

```diff
@@ -404,6 +404,14 @@ fn test_directory() {
     let mut file_names = dir_iter.map(|e| e.unwrap().file_name()).collect::<Vec<_>>();
     file_names.sort_unstable();
     assert_eq!(file_names, vec!["test_file_1", "test_file_2"]);
+    // Test that read_dir metadata calls succeed
+    assert_eq!(
+        &[true, true],
```

Comment:
Ideally we'd also make sure it returns `false` non directories.

Reply:
Sounds good, created https://github.com/rust-lang/rust/pull/103258